stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/composer/MessageComposerController.kt

```diff
@@ -686,17 +701,28 @@ public class MessageComposerController(
     }
 
     /**
-     * Shows cooldown countdown timer instead of send button when slow mode is enabled.
+     * Executes the message Call and shows cooldown countdown timer instead of send button
+     * when slow mode is enabled.
      */
-    private fun handleCooldownTimer() {
-        if (cooldownInterval > 0) {
+    private fun Call<Message>.enqueueAndHandleSlowMode() {
+        if (cooldownInterval > 0 && isSlowModeActive.value && !isInEditMode) {
             cooldownTimerJob?.cancel()
-            cooldownTimerJob = scope.launch {
-                for (timeRemaining in cooldownInterval downTo 0) {
-                    cooldownTimer.value = timeRemaining
-                    delay(OneSecond)
+
+            cooldownTimer.value = cooldownInterval
+            enqueue {
+                if (it.isSuccess) {
+                    cooldownTimerJob = scope.launch {
+                        for (timeRemaining in cooldownInterval downTo 0) {
+                            cooldownTimer.value = timeRemaining
+                            delay(OneSecond)
+                        }
+                    }
+                } else {
+                    cooldownTimer.value = 0
```

Comment:
Should we update the timer to 0 on that case?? :thinking: 
What happens in the case a user is offline, sends 100 messages and then comes back offline?
I think we don't really need to distinguish between success and failure here, whenever the `call` is executed, the timer should start

Reply:
Resolved - we reset the timer only if the user is online and the call failed (guidance from BE and mutual agreement).

This way in cases of NNBB and similar things we'll be able to resend/edit properly.

For editing - we don't disable the send UI if we're editing (cooldown doesn't affect edit)